src/components/molecules/Visualizer/hooks.ts

```diff
@@ -52,6 +62,25 @@ export default ({
 }) => {
   const engineRef = useRef<EngineRef>(null);
 
+  const [overriddenSceneProperty, overrideSceneProperty] = useState({});
+
+  const handleScenePropertyOverride = useCallback((property: any) => {
+    overrideSceneProperty(p => mergeProperty(p, property));
+  }, []);
```

Comment:
```suggestion
  const handleScenePropertyOverride = useCallback((property: any) => {
    overrideSceneProperty(property);
  }, []);
```

Reply:
That is how I originally had it. But that meant if, for example, I changed the tile type with one button, and then I changed the opacity with a slider widget, the changed tile type would revert back to the original. So only if the plugin remembers the changes, or does ALL changes at once, would multiple changes persist.
Do you have a better solution for that issue?

Reply:
@KaWaite OK, It is better to keep the override property for each plugin and merge them in one batch.

```ts
const [overriddenSceneProperty, overrideSceneProperty] = useState({});
```

->

```ts
const [overriddenSceneProperty, overrideSceneProperty] = useState<{ [pluginId: string]: any }>({});
```

---

```ts
const handleScenePropertyOverride = useCallback((property: any) => {
  overrideSceneProperty(property);
}, []);
```

->

```ts
const handleScenePropertyOverride = useCallback((pluginId: string, property: any) => {
  overrideSceneProperty(p => (pluginId && property ? { ...p, [pluginId]: property } : omit(p, pluginId));
}, []);
```

omit is lodash function

---

```ts
const mergedSceneProperty = useMemo(() => {
    return mergeProperty(sceneProperty, overriddenSceneProperty)
  }, [sceneProperty, overriddenSceneProperty]);
```

->

```ts
const mergedSceneProperty = useMemo(() => {
    return Object.values(overriddenSceneProperty).reduce((p, v) => mergeProperty(p, v), sceneProperty);
  }, [sceneProperty, overriddenSceneProperty]);
```